src/zenml/models/artifact_models.py

```diff
@@ -46,9 +46,13 @@ class ArtifactBaseModel(BaseModel):
     """Base model for artifacts."""
 
     name: str = Field(
-        title="Name of the output in the parent step.",
+        title="Name of the artifact.",
         max_length=STR_FIELD_MAX_LENGTH,
     )
```

Comment:
I would suggest storing pipeline_name and step_name separately, rather than combining them here, cause users might want to use `::` in their names and it might somehow affect the logic. Lookups by separate fields would also be a bit more readable, IMO.

Reply:
> cause users might want to use :: in their names and it might somehow affect the logic
Now that we also store `is_custom_name` it should be no problem to discern user-created names with `::` in it from the auto-generated ones, so I don't think this is an issue. 

Technically we actually have all this info in the DB already - the `ArtifactResponseModel` has a `producer_step_run_id` field which is the ID of the step that produced it (which in turn has a link to the run which has a link to the pipeline). So technically we wouldn't need to provide `name` in the request model at all. But we already had the field, so might as well use it 😄

TL;DR: Overall I prefer the current approach since it's the easiest and doesn't require additional DB columns